pages/sponsorship.js

```diff
@@ -0,0 +1,238 @@
+import Card, { CardHeader, CardTitle, CardContent, CardButtonContainer, CardDiv, TableDiv } from '../components/card'
+import styled from 'styled-components'
+import Button from '../components/button'
+import React, { useState } from 'react'
+import { EDIT, NEW } from '../constants'
+import fireDb, { db } from "../utility/firebase"
+
+
+// TODO: move font family higher, reduce need to redefine
+
+const Text = styled.text`
+    padding-right: 80px;
+    flex: 1;
+    white-space: nowrap;    
+    overflow: hidden;         
+    text-overflow: ellipsis;
+`
+const Actions = styled.div`
+    padding-right: 80px;
+    flex: 1;
+`
+
+class SponsorshipPage extends React.Component {
```

Comment:
I think we're using functional components for this project (it seems like the other PRs so far have been all functional components).

I know its a lot of work but we should stay consistent, could you switch this to a functional component?